### DISCUSSION TITLE
warn if URL previews are disabled on the server

### DISCUSSION THREAD

**Discussion by uhoreg** [element-hq/element-meta#834] [Ideas]
because it's confusing when you enable previews in the client, but still don't see previews

**Comment by turt2live** [comment]
The easiest way to check this is to do a couple URL preview tests. This is somewhat challenging though as the URLs may be inaccessible due to several reasons (firewall, lack of internet, geographic location, etc). It's also possible the homeserver has blocked the IP range (or only whitelisted a few), and most will have 127.0.0.1 blocked therefore checking the homeserver's URL itself is likely to fail.

**Comment by Nostradamos** [comment]
> The easiest way to check this is to do a couple URL preview tests. This is somewhat challenging though as the URLs may be inaccessible due to several reasons (firewall, lack of internet, geographic location, etc). It's also possible the homeserver has blocked the IP range (or only whitelisted a few), and most will have 127.0.0.1 blocked therefore checking the homeserver's URL itself is likely to fail.

Wouldn't it be better to implement some api to get to know that?

**Comment by turt2live** [comment]
I was trying to avoid that, as outdated servers could end up falsely saying they aren't enabled when they are. It would probably fit well into vector-im/element-web#5193

**Comment by Nostradamos** [comment]
Okay, got your point, thanks for clarification. Maybe add some backwards compatibility code for the beginning. But an api endpoint like needed for https://github.com/vector-im/riot-web/issues/5193 should be the way to solve this sort of problem in my opinion.

**Comment by lampholder** [comment]
I think an API is the way to go here - it should be possible to contrive "enabled/disabled/server couldn't/wouldn't tell me either way" sematics within said API so that older servers failure to report a setting or feature's being enabled != that feature being reported as disabled.

**Comment by lampholder** [comment]
And for the record I'm assuming we're warning on the settings page, or disabling the setting(s) rather than popping up 'this server does not support url previews' with every url rendered in the timeline :)
